### PR TITLE
[DK, FD][APB-901] decode error messages, refactor, add more tests

### PR DESCRIPTION
**Pull request by dimitraka71**

### CODE REVIEW

File: app/uk/gov/hmrc/agentsubscriptionfrontend/models/Address.scala [hunk 1]
```diff
@@ -137,7 +136,4 @@ object Address{
 
     OFormat[Address](reads, formatAddressValue)
   }
-
-  val renderErrors: PostCodeError => String = (errors: PostCodeError) => errors.foldLeft("")(_+", "+_).substring(1)
```
Comment (frankoid) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/models/Address.scala]: Does this have to be a `val`, can it be a `def` instead? `def` is less advanced -> easier to understand

Comment (frankoid) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/models/Address.scala]: Is there a unit test for this? If not please add one

**Comment by frankoid** [comment]
`sbt clean test it:test` fails for me, does it pass for you?

File: app/uk/gov/hmrc/agentsubscriptionfrontend/models/Address.scala (hunk 1 above)
Comment (dimitraka71) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/models/Address.scala]: according to the renderErrors anonymous function: this is the updated, https://github.com/hmrc/agent-subscription-frontend/pull/78/commits/197c51130aac51d8e9dbef15f2fe731fa42f4be6, (commit 197c51130aac51d8e9dbef15f2fe731fa42f4be6). and all the tests are passed

Comment (frankoid) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/models/Address.scala]: I think my comments still apply - where is the unit test for `renderErrors` and does it need to be a `val`?

Comment (dimitraka71) [inline app/uk/gov/hmrc/agentsubscriptionfrontend/models/Address.scala]: I can write unit tests for this anon function. It has the same functionality with def

**Pull request closed without merging**